### PR TITLE
Faster and more robust checkbox test

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

Test for a selected product can be made faster (do not search in the page twice) and more robuts (current test fails in certain conditions).

## Links

Ports:
* 4.1: SUSE/spacewalk#13381
* 4.0: SUSE/spacewalk#13383

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

### CODE REVIEW

**Review by ktsamis — APPROVED** [review]
LGTM

**Review by lkotek — APPROVED** [review]
Great, thanks :-)

**Review by calancha — APPROVED** [review]

File: testsuite/features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -511,8 +511,9 @@
 
 Then(/^I should see the "(.*?)" selected$/) do |product|
   xpath = "//span[contains(text(), '#{product}')]/ancestor::div[contains(@class, 'product-details-wrapper')]"
-  product_identifier = find(:xpath, xpath)['data-identifier']
-  raise "#{product_identifier} is not checked" unless has_checked_field?('checkbox-for-' + product_identifier)
+  product = find(:xpath, xpath)
+  selected = product.find(:xpath, "./div/input[@type='checkbox']").checked?
+  raise "#{product['data-identifier']} is not checked" unless selected
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:516]:
```suggestion
within(xpath) do
  raise "#{product['data-identifier']} is not checked" unless product.find(:xpath, "./div/input[@type='checkbox']").checked?
end
```

Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:516]: why not, it's more elegant as you suggest

BUT `product` is a node, what you want in the error message is `product['data-identifier']`, a string

Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb]: Done.

Comment (srbarrios) [reply testsuite/features/step_definitions/common_steps.rb]: @Bischoff it is impossible that product is undefined, we are passing a value as a parameter in the step definition.
With your extra change, we are adding an extra find, anyhow, as it is only called on a raise, fine with this approach.

**Pull request merged**